openmmml/models/macepotential.py

```
@@ -0,0 +1,222 @@
+
+
+                if indices is None:
+                    self.indices = None
+                else:
+                    self.indices = torch.tensor(indices, dtype=torch.int64)
+
+                if interaction_energy is True:
+                    self.return_energy_type = "interaction_energy"
+                else:
+                    self.return_energy_type = "energy"
+
+            def forward(self, positions, boxvectors: Optional[torch.Tensor] = None):
+                
+                # setup positions
+                positions = positions.to(self.default_dtype)
+                if self.indices is not None:
+                    positions = positions[self.indices]
+
+                positions = positions*self.nm_to_distance
+
+                if boxvectors is not None:
+                    cell = boxvectors.to(self.default_dtype) * self.nm_to_distance
+                    pbc = True
+                else:
+                    cell = torch.eye(3, device=positions.device)
+                    pbc = False
+  
+                mapping, shifts_idx = simple_nl(positions, cell, pbc, self.model.r_max)
+                
+                edge_index = torch.stack((mapping[0], mapping[1]))
+
+                shifts = torch.mm(shifts_idx, cell)
+
+                # create input dict
+                input_dict = { "ptr" : self.ptr, 
+                              "node_attrs": self.node_attrs, 
+                              "batch": self.batch, 
+                              "pbc": self.pbc,
+                              "cell": cell,
+                              "positions": positions,
+                              "edge_index": edge_index,
+                              "unit_shifts": shifts_idx,
+                              "shifts": shifts}
+                
+                # predict
+                out = self.model(input_dict,compute_force=False)
+
+                energy = out[self.return_energy_type]
+                if energy is None:
```

How can you get energy `None`? The simulation should fail rather than silently assume it is 0.